presto-hive/src/main/java/com/facebook/presto/hive/s3/PrestoS3ClientFactory.java

```diff
@@ -65,14 +65,33 @@ public class PrestoS3ClientFactory
     private static String s3UserAgentSuffix = "presto";
 
     @GuardedBy("this")
-    private AmazonS3 s3Client;
+    private S3Client s3Client;
 
-    synchronized AmazonS3 getS3Client(Configuration config, HiveClientConfig clientConfig)
+    @GuardedBy("this")
+    private S3AsyncClient s3AsyncClient;
+
+    synchronized S3Client getS3Client(Configuration config, HiveClientConfig clientConfig)
     {
         if (s3Client != null) {
             return s3Client;
         }
 
+        s3Client = buildS3Client(config, clientConfig);
+        return s3Client;
+    }
```

Comment:
As PrestoS3SelectClient uses `selectObjectContent` and in V2, 'selectObjectContent()' is available only on the S3AsyncClient, I don't think we need this anymore, as this class `PrestoS3ClientFactory` is only used by `PrestoS3SelectClient`.